bindings/cpp/session.cpp

```diff
@@ -2233,12 +2248,14 @@ error_info session::mix_states(const key &id, std::vector<int> &groups)
 }
 
 async_iterator_result session::start_iterator(const key &id, const std::vector<dnet_iterator_range>& ranges,
-								uint32_t type, uint64_t flags,
-								const dnet_time& time_begin, const dnet_time& time_end)
+		uint32_t type, uint64_t flags, const dnet_time& time_begin, const dnet_time& time_end)
 {
-	auto ranges_size = ranges.size() * sizeof(ranges.front());
+	size_t ranges_size = 0;
+	if (ranges.size() != 0)
+		ranges_size = ranges.size() * sizeof(ranges.front());
```

Comment:
From C++14 standard:
5.3.3.1:
The sizeof operator yields the number of bytes in the object representation of its operand. The operand is either an expression, which is an unevaluated operand (Clause 5), or a parenthesized type-id.